Libraries/LibGUI/ColorPicker.cpp

```diff
@@ -28,88 +28,278 @@
+#include <LibGUI/Label.h>
 #include <LibGUI/SpinBox.h>
-#include <LibGUI/Widget.h>
+#include <LibGUI/TextBox.h>
 #include <LibGfx/Palette.h>
 
 namespace GUI {
 
-ColorPicker::ColorPicker(Color color, Window* parent_window)
+ColorPicker::ColorPicker(Color color, Window* parent_window, String title)
     : Dialog(parent_window)
     , m_color(color)
 {
-    set_title("Edit Color");
-    build();
+    set_icon(Gfx::Bitmap::load_from_file("/res/icons/16x16/color-chooser.png"));
+
+    set_title(title);
+
+    m_current_screen = ScreenList::Palette;
+    build_ui();
 }
 
 ColorPicker::~ColorPicker()
 {
 }
 
-void ColorPicker::build()
+void ColorPicker::build_ui()
 {
-    auto& horizontal_container = set_main_widget<Widget>();
-    horizontal_container.set_fill_with_background_color(true);
-    horizontal_container.set_layout<HorizontalBoxLayout>();
-    horizontal_container.layout()->set_margins({ 4, 4, 4, 4 });
+    auto& root_container = set_main_widget<Widget>();
+    root_container.set_layout<VerticalBoxLayout>();
+    root_container.layout()->set_margins({ 4, 4, 4, 4 });
+    root_container.set_fill_with_background_color(true);
+
+    m_manin_container = root_container.add<Widget>();
+    m_manin_container->set_size_policy(SizePolicy::Fill, SizePolicy::Fill);
+    m_manin_container->set_layout<VerticalBoxLayout>();
+    m_manin_container->layout()->set_spacing(4);
+
+    auto& button_container = root_container.add<Widget>();
+    button_container.set_size_policy(SizePolicy::Fill, SizePolicy::Fixed);
+    button_container.set_preferred_size(0, 22);
+    button_container.set_layout<HorizontalBoxLayout>();
+    button_container.layout()->set_spacing(4);
+    button_container.layout()->add_spacer();
+
+    m_custom_button = button_container.add<Button>();
+    m_custom_button->set_size_policy(SizePolicy::Fixed, SizePolicy::Fill);
+    m_custom_button->set_preferred_size(80, 0);
+    m_custom_button->set_text("Custom Color");
+    m_custom_button->on_click = [this] {
+        if (m_current_screen == ScreenList::Palette) {
+            this->build_ui_custom(*m_manin_container);
+        } else {
+            this->build_ui_palette(*m_manin_container);
+        }
+    };
+
+    auto& cancel_button = button_container.add<Button>();
+    cancel_button.set_size_policy(SizePolicy::Fixed, SizePolicy::Fill);
+    cancel_button.set_preferred_size(80, 0);
+    cancel_button.set_text("Cancel");
+    cancel_button.on_click = [this] {
+        done(ExecCancel);
+    };
+
+    auto& ok_button = button_container.add<Button>();
+    ok_button.set_size_policy(SizePolicy::Fixed, SizePolicy::Fill);
+    ok_button.set_preferred_size(80, 0);
+    ok_button.set_text("Select");
+    ok_button.on_click = [this] {
+        done(ExecOK);
+    };
+
+    if (m_current_screen == ScreenList::Palette) {
+        build_ui_palette(*m_manin_container);
+    } else {
+        build_ui_custom(*m_manin_container);
+    }
+}
+
+void ColorPicker::build_ui_palette(Widget& kok)
```

Comment:
In the header you are using `root_container` instead of `kok` for the name of this argument. 

```suggestion
void ColorPicker::build_ui_palette(Widget& root_container)
```